### PR TITLE
feat: make context stateful and add dataSuffix option

### PR DESCRIPTION
**Pull request by wilsoncusack**
## Description

In this PR I update mintDialogContext to be stateful, so that certain of its values can be set within the dialog flow, and also add a dataSuffix to the mint which is only used for third web currently. 

## Deploy Notes

Notes regarding deployment of the contained body of work. These should note any
db migrations, nginx routes, infrastructure changes, and anything that must be done before deployment.

- [ ] Need to add environment variables
  - `ENV_VAR=`

### CODE REVIEW

File: src/components/MintDialog/elements/NativeMintButton.tsx [hunk 1]
```diff
@@ -53,43 +55,51 @@ export const NativeMintButton: FC<NativeMintButtonProps> = ({
...
+            return
+          }
+          const {request} = await publicClient.simulateContract({
+            abi: tw721ABI,
+              address: address as Address,
+              functionName: 'claim',
+              args: [
+                account,
+                BigInt(quantity),
+                '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE',
```
Comment (zencephalon) [inline src/components/MintDialog/elements/NativeMintButton.tsx:83]: Can we name this magic constant?

File: src/components/MintDialog/elements/NativeMintButton.tsx [hunk 2]
```diff
@@ -53,43 +55,51 @@ export const NativeMintButton: FC<NativeMintButtonProps> = ({
...
+                price / BigInt(quantity),
+                {
+                  proof: [],
+                  quantityLimitPerWallet: 2n ** 256n - 1n,
```
Comment (zencephalon) [inline src/components/MintDialog/elements/NativeMintButton.tsx:87]: This seems worth naming too

**Review by zencephalon — APPROVED** [review]

**Comment by cb-heimdall** [comment]
Review Error for zencephalon @ 2023-08-12 22:50:34 UTC
User failed mfa authentication, see [go/mfa-help](http://go/mfa-help)

**Review by zencephalon — APPROVED** [review]

**Comment by cb-heimdall** [comment]
Review Error for zencephalon @ 2023-08-12 22:57:10 UTC
User failed mfa authentication, see [go/mfa-help](http://go/mfa-help)

**Review by zencephalon — APPROVED** [review]

**Comment by cb-heimdall** [comment]
Review Error for zencephalon @ 2023-08-12 22:57:29 UTC
User failed mfa authentication, see [go/mfa-help](http://go/mfa-help)

File: src/components/MintDialog/elements/NativeMintButton.tsx (hunk 1 above)
Comment (wilsoncusack) [reply src/components/MintDialog/elements/NativeMintButton.tsx:83]: agree!

File: src/components/MintDialog/elements/NativeMintButton.tsx (hunk 2 above)
Comment (wilsoncusack) [reply src/components/MintDialog/elements/NativeMintButton.tsx:87]: yeah agree, this is all a bit jank here. Was just moving this into a new function call as the wagmi cli generated stuff was dropping the datasuffix

**Review by ximxim — APPROVED** [review]

**Comment by cb-heimdall** [comment]
Review Error for ximxim @ 2023-08-14 00:00:28 UTC
User failed mfa authentication, see [go/mfa-help](http://go/mfa-help)

**Review by mdehoog — APPROVED** [review]

**Pull request merged**